llvm/lib/Target/AMDGPU/AMDGPULateCodeGenPrepare.cpp

```diff
@@ -365,13 +365,33 @@ bool LiveRegOptimizer::optimizeLiveType(
       else
         MissingIncVal = true;
     }
-    Instruction *DeadInst = Phi;
     if (MissingIncVal) {
-      DeadInst = cast<Instruction>(ValMap[Phi]);
-      // Do not use the dead phi
-      ValMap[Phi] = Phi;
+      Value *DeadVal = ValMap[Phi];
+      // The coercion chain of the PHI is broken. Delete the Phi
+      // from the ValMap and any connected / user Phis.
```

Comment:
Isn't this what ValueMap is supposed to take care of for you? The ValueHandles should have been updated when you replace the value?

Reply:
Not quite sure how we could more easily accomplish this using ValueHandles -- We never RAUW so by simply replacing to use ValueHandles we would still be where we are. 

But if we changed the implementation of PHI node handling to use RAUW, it seems to me that we would still create broken PHIs and would need some unwinding mechanism.

Reply:
But if you do RAUW, the weak handles handle the case where the value is ultimately deleted